client/src/components/rounds/Show.tsx

```diff
@@ -1,95 +1,104 @@
 import { useEffect, useState } from "react";
 import { Link, useParams } from "react-router-dom";
-import { shallowEqual, useSelector, useDispatch } from "react-redux";
-import { RootState } from "../../reducers";
+import { useNetwork } from "wagmi";
+import {
+  getRoundApplicationMetadata,
+  getRoundMetadata,
+} from "../../actions/rounds";
+import { useClients } from "../../hooks/useDataClient";
+import useLocalStorage from "../../hooks/useLocalStorage";
 import { roundApplicationPath } from "../../routes";
-import { loadRound, unloadRounds } from "../../actions/rounds";
-import { Status } from "../../reducers/rounds";
-import { networkPrettyName } from "../../utils/wallet";
+import { useFetchRoundByAddress } from "../../services/graphqlClient";
 import { formatDate } from "../../utils/components";
+
 import Button, { ButtonVariants } from "../base/Button";
+import TextLoading from "../base/TextLoading";
 
 function Round() {
+  const [loading, setLoading] = useState(true);
   const [roundData, setRoundData] = useState<any>();
-
+  const [roundToApply, setRoundToApply] = useLocalStorage("roundToApply", null);
+  const { chain } = useNetwork();
   const params = useParams();
-  const dispatch = useDispatch();
 
   const { roundId, chainId } = params;
 
-  const props = useSelector((state: RootState) => {
-    const roundState = state.rounds[roundId!];
-    const status = roundState ? roundState.status : Status.Undefined;
-    const error = roundState ? roundState.error : undefined;
-    const round = roundState ? roundState.round : undefined;
-    const web3ChainId = state.web3.chainID;
-    const roundChainId = Number(chainId);
-
-    return {
-      roundState,
-      status,
-      error,
-      round,
-      web3ChainId,
-      roundChainId,
-    };
-  }, shallowEqual);
+  const { roundManagerClient } = useClients(Number(chainId));
 
-  useEffect(() => {
-    if (roundId !== undefined) {
-      dispatch(unloadRounds());
-      dispatch(loadRound(roundId));
+  async function fetchRound() {
+    if (!roundManagerClient) return;
+    const roundInfo = await useFetchRoundByAddress(
+      roundManagerClient,
+      roundId!
+    );
+
+    if (!roundInfo) {
+      console.error("Cannot load round", roundId);
+      return;
     }
-  }, [dispatch, roundId]);
-
-  if (props.web3ChainId !== props.roundChainId) {
```

Comment:
I added this a couple days ago to indicate that the round contract is on a different network than the one the user is connected to https://github.com/gitcoinco/grant-hub/pull/243 Any chance we could add this back in?

Reply:
Yep, on it!